packages/js/src/insights/initializer.js

```diff
@@ -44,8 +44,8 @@ const createUpdater = () => {
  * @returns {function} The subscriber.
  */
 const createSubscriber = () => {
-	const { getEditorDataContent, getContentLocale } = select( "yoast-seo/editor" );
-	const collector = createCollector( getEditorDataContent, getContentLocale );
+	const { getContentLocale } = select( "yoast-seo/editor" );
+	const collector = createCollector( getContentLocale, collectData );
```

Comment:
Here, we watch the changes in the content including the modifications: `collectData` function takes care of retrieving the content including the modifications (e.g. modifications/content from the ACF fields).
This way, when there is a change in the ACF field for example, we would recalculate the analysis for Insights immediately without the need to save the changes first.

For context: `getEditorDataContent` only retrieves the content from the main text editor. This is why we previously didn't update the analysis result in Insights when there is a change coming from a non-main text editor, like an ACF field.

Reply:
Great find! 💯

Reply:
As discussed on Slack, let's offload this fix to a new pull request! 🚀

Reply:
Done, https://github.com/Yoast/wordpress-seo/pull/20845 has been merged :octocat: